drracket-test/tests/drracket/populate-compiled.rkt

```diff
@@ -1,15 +1,15 @@
 #lang racket/base
-(require racket/file
-         racket/system
-         compiler/find-exe
-         pkg/lib)
+(require compiler/find-exe
+         pkg/lib
+         racket/file
+         racket/system)
 
```

Comment:
Is the reordering above to make it alphabetical? I generally go out of my way to put more "core" libraries first, so I would have written the original here instead of the sorted version.

Reply:
Yeah. It's attempting to use the same ordering that `racket-mode`'s "tidy requires" command uses. I figured keeping all the various different bits of automation in sync on this is desirable.

Reply:
What if we choose the following order:
- `racket/base`
- libraries included in `racket`
- libraries that are in `racket/`
- everything else

with alphabetical order within each group?

And I agree that such a change should be synchronized with `racket-mode`.

Reply:
We could. There's a few downsides:

- This logic needs to exist in Resyntax, `racket-mode`, and any other tools that want to edit `require` forms.
- That logic has to be updated whenever there's a change to the list of libraries included in `racket`. (The `racket/` libraries can just be identified textually.) There _might_ be a way to figure this out by loading `racket` and seeing what it re-exports, but I'm not sure that will work for tools that want to tidy `require` forms without expanding macros or loading other modules.
- Making `racket` re-export a library that's already in use would change everyone's import orders.

And as I understand it, the upside is that when a human reads through the list of `require`'d modules, core libraries show up first.

Personally I'm not really sold on the idea that this benefit is worth the effort. I greatly enjoy living in a world where I can just not care about the order here and slap in a module wherever when I need to import something new and let the tools keep it manageable.

cc @greghendershott for any relevant `racket-mode` context

Reply:
Treating "libraries included in `racket`" specially seems like the biggest complication. What about an order like this?

1. `^racket(/*)?$` libraries
2. other libraries imported by symbolic name
3. relative-path libraries, re https://github.com/racket/drracket/pull/681#discussion_r1765066074 (I also put these last, and my vague impression is that's common)

On that last point, how do Resyntax and `racket-mode` currently handle the various non-symbolic `require` subforms?

> And as I understand it, the upside is that when a human reads through the list of `require`'d modules, core libraries show up first.
> 
> Personally I'm not really sold on the idea that this benefit is worth the effort. I greatly enjoy living in a world where I can just not care about the order here and slap in a module wherever when I need to import something new and let the tools keep it manageable.

I usually organize `require`s vaguely thematically, e.g. modules for IO, modules for macro support, etc., which is not going to be legible to tooling. On the other hand, I do see the advantage of just letting the tooling handle the ordering.

Reply:
@LiberalArtist That would be much more doable. I'm ambivalent on its value (since all the `racket/` imports get grouped together when sorted alphabetically anyway) but I'm open to being outvoted.

> On that last point, how do Resyntax and `racket-mode` currently handle the various non-symbolic `require` subforms?

I tried to make Resyntax follow the logic written down in the [`racket-tidy-requires`](https://racket-mode.com/#racket_002dtidy_002drequires) documentation. Although... I now realize I messed up the ordering of `(only-in ...)` and friends. Whoops. Filed jackfirth/resyntax#306.

Reply:
I spoke with Matthew and Robby about this offline and we concluded that moving `racket/` modules to the top isn't worth the effort. Matthew's concern came from his work in Rhombus where he's doing that all over the place, but he's also putting things like the `syntax/` modules at the top with the core libraries. So a purely textual approach wouldn't work for him. We agreed to try the existing behavior for the time being and reevaluate later if needed.